### ISSUE TITLE
EmptyError: no elements in sequence

### ISSUE THREAD

**Issue by syeddsherazi** [shlomiassaf/angular2-modal#178]
I'm using Angular 2 RC5 with webpack. Upon clicking on "Yes" in modal, I'm calling a function "deletecourse", in delete course I delete the course from DB and navigate to some path. Below are the snippets of code.

Dialog:
  ```
  showConfirm(ev) {
        let dialog = this.modal.confirm()
            .size('sm')
            .isBlocking(true)
            .showClose(false)
            .title('Alert')
            .body('Are you sure you want to delete this course?')
            .okBtn('Yes')
            .cancelBtn('No')
            .open();
        dialog.then(dialog => dialog.result).then((dialog) => {
            this.deleteCourse();
        }).catch(dialog => { });
    };
```

Delete Course:
```

 deleteCourse() {
        this.courseService.deleteCourse(this.selectedId).subscribe((result) => {
            if (result.success === '1') {
                this.router.navigate(['courses']);
            }
        });
    }
```

Now the Service is working fine. It deletes the course. The issue comes on navigate. If I remove router.navigate then it works fine. But on router navigation, it doesn't go to the passed route and gives following error.

**EmptyError: no elements in sequence**

**Comment by noorahmed94** [comment]
facing the same issue, @shlomiassaf any specific reason for it?

**Comment by fahad-nxvt** [comment]
Any update on this?

**Comment by noorahmed94** [comment]
@shlomiassaf is it something related to navigating on dialog close event because i am also getting same error as @fahad-nxvt  on dialog close

**Comment by shlomiassaf** [comment]
@noorahmed94 @fahad-nxvt @syeddsherazi This is probably due to animation trying to execute but fails.

As a workaround try to call `router.navigate` inside `setTimeout` and let me know how it goes.
If it solves the issue I might need to defer the resolve, but I need to see first if this is the case.

**Comment by shlomiassaf** [comment]
@syeddsherazi This might be fixed with #203 tough I think it's a different issue.

I didn't get a response for my workaround suggestion so I will assume it helped since the issue became stale.

If it's not solved please open a new issue.
